LibOS/shim/src/bookkeep/shim_signal.c

```diff
@@ -407,8 +416,46 @@ static void illegal_upcall (PAL_PTR event, PAL_NUM arg, PAL_CONTEXT * context)
         if (context)
             debug("illegal instruction at 0x%08lx\n", context->IP);
 
-        deliver_signal(ALLOC_SIGINFO(SIGILL, ILL_ILLOPC,
-                                     si_addr, (void *) arg), context);
+        uint8_t * rip = (uint8_t*)context->IP;
```

Comment:
Sorry to comment on a merged PR, but is `context` always valid here? There is a check for `if (context)` above, which suggests it may be NULL...